### PR TITLE
Fix list on password policies to list the policies containing slashes

### PR DESCRIPTION
**Pull request by akshya96**
Issue: https://github.com/hashicorp/vault/issues/17117
Slack conversation: https://hashicorp.slack.com/archives/C0287E435NE/p1694714032988359
Jira: https://hashicorp.atlassian.net/browse/VAULT-20052

Vault allows us to create password policies with "/" similar to other policy types like acl
Example: vault write sys/policies/password/example/example1 policy=@example_policy.hcl is allowed 

but the list on password policies does not list the values after "/"
Example response: {"request_id":"9b1eadda-0511-02e0-f17f-7b9fe1d2c8b7","lease_id":"","renewable":false,"lease_duration":0,"data":{"keys":["example/"]},"wrap_info":null,"warnings":null,"auth":null}

Making this work consistently with acl policies. This fix produces the following response:
{"request_id":"9b1eadda-0511-02e0-f17f-7b9fe1d2c8b7","lease_id":"","renewable":false,"lease_duration":0,"data":{"keys":["example/example1"]},"wrap_info":null,"warnings":null,"auth":null}

### CODE REVIEW

**Review by biazmoreira — APPROVED** [review]

**Review by mpalmi — COMMENTED** [review]
Looks good! Just a couple of things I'd like to see before we merge this.

File: vault/logical_system.go [hunk 1]
```diff
@@ -3210,12 +3210,14 @@ const (
 )
 
 // handlePoliciesPasswordList returns the list of password policies
-func (*SystemBackend) handlePoliciesPasswordList(ctx context.Context, req *logical.Request, data *framework.FieldData) (resp *logical.Response, err error) {
-	keys, err := req.Storage.List(ctx, "password_policy/")
+func (b *SystemBackend) handlePoliciesPasswordList(ctx context.Context, req *logical.Request, data *framework.FieldData) (resp *logical.Response, err error) {
```
[the remark this answers is not in the document]
Comment (akshya96) [reply vault/logical_system.go:3213]: Yes, good catch! I added it while trying something else and forgot to remove it.

File: vault/logical_system_test.go [hunk 2]
```diff
@@ -4739,6 +4739,26 @@ func TestHandlePoliciesPasswordList(t *testing.T) {
 				},
 			},
 		},
+		"policy with /": {
```
[the remark this answers is not in the document]
Comment (akshya96) [reply vault/logical_system_test.go:4742]: Added test for these two cases listed above

**Review by mpalmi — APPROVED** [review]
Looks great!

**Pull request merged**